### PR TITLE
Chefstyle updates

### PR DESCRIPTION
**Pull request by lamont-granquist**
These are new proposed cops

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-core thoughts and feels?

**Comment by thommay** [comment]
I dislike EmptyCaseCondition but don't really care.

I find MultilineArrayBraceLayout a step backwards in terms of readability.

**Comment by lamont-granquist** [comment]
so @thommay the default for MultilineArrayBraceLayout is 'symmetrical' and supports either of these:

```ruby
# symmetrical: good
# new_line: bad
# same_line: good
[ :a,
  :b ]

# symmetrical: good
# new_line: good
# same_line: bad
[
  :a,
  :b
]
```

i think the problem you're seeing is that you don't like that the opening brace did not start on its own line in those cases.

**Comment by thommay** [comment]
ok, I mean the first example is awful (which is what you had in this PR)

**Comment by lamont-granquist** [comment]
right but that's a function of the first line being written incorrectly, MultilineArrayBraceLayout is behaving well, we just need to get around to the cop that will fix the first line.  if you look at the code it is fixing it is worse:

```
[ :a,
  :b,
]
```

that is something nobody wants.

**Comment by lamont-granquist** [comment]
and I think I wrote both of the code fragments that EmptyCaseCondition caught.  i'm real tempted to drop that one.

File: lib/chef/exceptions.rb [hunk 1]
```diff
@@ -76,10 +76,10 @@ class User < RuntimeError; end
     class Group < RuntimeError; end
     class Link < RuntimeError; end
     class Mount < RuntimeError; end
-    class Reboot < Exception; end
```
Comment (danielsdeleo) [inline lib/chef/exceptions.rb]: Seems like we may have made this inherit from `Exception` intentionally?

**Comment by danielsdeleo** [comment]
I'm not sure we want to auto-fix the `InheritException` ones. In that case I'm okay with enabling it and using the magic comments to disable for specific cases where we know it's what we want.

**Comment by lamont-granquist** [comment]
EmptyCaseCondition gone
MultilineArrayBraceLayout gone (too much trouble)

**Comment by lamont-granquist** [comment]
@chef/client-core okay fixes applied i think

**Comment by danielsdeleo** [comment]
Looks ok to me now

**Pull request merged**